test/core/surface/lame_client_test.cc

```diff
@@ -114,8 +114,7 @@ TEST(LameClientTest, MainTest) {
                                 tag(1), nullptr);
   ASSERT_EQ(GRPC_CALL_OK, error);
 
-  /* the call should immediately fail */
-  cqv.Expect(tag(1), false);
+  cqv.Expect(tag(1), grpc_core::CqVerifier::AnyStatus{});
```

Comment:
Similar to the case above: Let's try to find a better story here, so that we can have consistent behavior between the two stacks, or at least expect the right thing for each stack explicitly.

Reply:
done

Reply:
Similar comment here as in cancel_before_invoke.cc: Is there a reasonable way to check which stack we're using and assert the right value in each case?

Reply:
latest code has something here that does assert based on stack